plugins/common/tls/utils.go

```diff
@@ -36,3 +43,58 @@ func ParseTLSVersion(version string) (uint16, error) {
 	sort.Strings(available)
 	return 0, fmt.Errorf("unsupported version %q (available: %s)", version, strings.Join(available, ","))
 }
+
+func readFile(filename string) []byte {
+	octets, err := os.ReadFile(filename)
+	if err != nil {
+		panic(fmt.Sprintf("reading %q: %v", filename, err))
+	}
+	return octets
+}
+
+func ReadCertificate(filename string) string {
```

Comment:
We're returning `string` here only to cast it to `[]byte` in `loadCertificate()` when it calls `tls.X509KeyPair()`. I think it would be best to just return []byte here. We should also be returning error here and handling that in loadCertificate().